File: client/mobile/app/src/main/java/com/example/fitsforkip/data/model/IncidentHistory.java
package com.example.fitsforkip.data.model;

import java.util.Date;

public class IncidentHistory {
    private int incidentId;
    private Integer equipmentId;
    private Date startTime;
    private Date endTime;
    private Double duration;
    private Integer typeId;
    private String reason;
    private String solution;
    private String issue;
    private String status;
    private Date createdDate;
    private String reportedByUserId;
    private String assignedTo;
    private boolean isTechSupport;

    // Constructors
    public IncidentHistory() {}

    public IncidentHistory(int incidentId, Integer equipmentId, Date startTime, Date endTime, Double duration, Integer typeId, String reason, String solution, String issue, String status, Date createdDate, String reportedByUserId, String assignedTo, boolean isTechSupport) {
        this.incidentId = incidentId;
        this.equipmentId = equipmentId;
        this.startTime = startTime;
        this.endTime = endTime;
        this.duration = duration;
        this.typeId = typeId;
        this.reason = reason;
        this.solution = solution;
        this.issue = issue;
        this.status = status;
        this.createdDate = createdDate;
        this.reportedByUserId = reportedByUserId;
        this.assignedTo = assignedTo;
        this.isTechSupport = isTechSupport;
    }

    // Getters and setters
    public int getIncidentId() { return incidentId; }
    public void setIncidentId(int incidentId) { this.incidentId = incidentId; }

    public Integer getEquipmentId() { return equipmentId; }
    public void setEquipmentId(Integer equipmentId) { this.equipmentId = equipmentId; }

    public Date getStartTime() { return startTime; }
    public void setStartTime(Date startTime) { this.startTime = startTime; }

    public Date getEndTime() { return endTime; }
    public void setEndTime(Date endTime) { this.endTime = endTime; }

    public Double getDuration() { return duration; }
    public void setDuration(Double duration) { this.duration = duration; }

    public Integer getTypeId() { return typeId; }
    public void setTypeId(Integer typeId) { this.typeId = typeId; }

    public String getReason() { return reason; }
    public void setReason(String reason) { this.reason = reason; }

    public String getSolution() { return solution; }
    public void setSolution(String solution) { this.solution = solution; }

    public String getIssue() { return issue; }
    public void setIssue(String issue) { this.issue = issue; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    public Date getCreatedDate() { return createdDate; }
    public void setCreatedDate(Date createdDate) { this.createdDate = createdDate; }

    public String getReportedByUserId() { return reportedByUserId; }
    public void setReportedByUserId(String reportedByUserId) { this.reportedByUserId = reportedByUserId; }

    public String getAssignedTo() { return assignedTo; }
    public void setAssignedTo(String assignedTo) { this.assignedTo = assignedTo; }

    public boolean isTechSupport() { return isTechSupport; }
    public void setTechSupport(boolean isTechSupport) { this.isTechSupport = isTechSupport; }
}

File: FITSKIP.Domain/Entities/IncidentHistory.cs
using System;
using System.Collections.Generic;

namespace FITSKIP.Domain.Entities;

public partial class IncidentHistory
{
    public int IncidentId { get; set; }

    public int? EquipmentId { get; set; }

    public int? LineId { get; set; }

    public DateTime? StartTime { get; set; }

    public DateTime? EndTime { get; set; }

    public decimal? Duration { get; set; }

    public int? TypeId { get; set; }

    public string? Reason { get; set; }

    public string? Solution { get; set; }

    public string? Issue { get; set; }

    public string? ImageUrl { get; set; }

    public string? Status { get; set; } = "Chờ xử lý"; // Trạng thái: Chờ xử lý, Đang xử lý, Hoàn thành, Hủy

    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;

    public string? ReportedByUserId { get; set; }

    public string? AssignedTo { get; set; }

    public bool IsTechSupport { get; set; } = false;

    public virtual Equipment? Equipment { get; set; }

    public virtual Line? Line { get; set; }

    public virtual StopType? Type { get; set; }

    public virtual User? ReportedByUser { get; set; }

    public virtual ICollection<IncidentShift> IncidentShifts { get; set; } = new List<IncidentShift>();
}
